spec/miscellaneous-updates.html

```diff
@@ -0,0 +1,567 @@
+      1. Let _hasSet_ be ? HasProperty(_Obj_, *"set"*).
+      1. If _hasSet_ is *true*, then
+        1. Let _setter_ be ? Get(_Obj_, *"set"*).
+        1. If IsCallable(_setter_) is *false* and _setter_ is not *undefined*, throw a *TypeError* exception.
+        1. Set _desc_.[[Set]] to _setter_.
+      1. If _desc_.[[Get]] is present or _desc_.[[Set]] is present, then
+        1. If _desc_.[[Value]] is present or _desc_.[[Writable]] is present, throw a *TypeError* exception.
+      1. Return _desc_.
+    </emu-alg>
+  </emu-clause>
+
+  <emu-clause id="sec-createlistfromarraylike" type="abstract operation">
+    <h1>
+      CreateListFromArrayLike (
+        _obj_: unknown,
+        optional _elementTypes_: a List of names of ECMAScript Language Types,
+      )
+    </h1>
+    <emu-note type="editor">
+      These changes allow using a Tuple for the arguments list passed to `Function.prototype.apply`, `Reflect.apply` and `Reflect.construct`.
+    </emu-note>
+    <emu-alg>
+      1. If _elementTypes_ is not present, set _elementTypes_ to &laquo; Undefined, Null, Boolean, String, Symbol, Number, BigInt, <ins>Record, Tuple, Box</ins> Object &raquo;.
+      1. <ins>If Type(_obj_) is Tuple, set _obj_ to ! ToObject(_obj_).</ins>
+      1. <ins>Else</ins> if Type(_obj_) is not Object, throw a *TypeError* exception.
+      1. Let _len_ be ? LengthOfArrayLike(_obj_).
+      1. Let _list_ be a new empty List.
+      1. Let _index_ be 0.
+      1. Repeat, while _index_ &lt; _len_,
+        1. Let _indexName_ be ! ToString(𝔽(_index_)).
+        1. Let _next_ be ? Get(_obj_, _indexName_).
+        1. If Type(_next_) is not an element of _elementTypes_, throw a *TypeError* exception.
+        1. Append _next_ as the last element of _list_.
+        1. Set _index_ to _index_ + 1.
+      1. Return _list_.
+    </emu-alg>
+  </emu-clause>
+
+  <emu-clause id="sec-add-entries-from-iterable" type="abstract operation">
+    <h1>
+      AddEntriesFromIterable (
+        _target_: unknown,
+        _iterable_: an ECMAScript language value, but not *undefined* or *null*,
+        _adder_: a function object,
+      )
+    </h1>
+    <dl class="header">
+      <dt>description</dt>
+      <dd>_adder_ will be invoked, with _target_ as the receiver.</dd>
+    </dl>
+    <emu-alg>
+      1. If IsCallable(_adder_) is *false*, throw a *TypeError* exception.
+      1. Let _iteratorRecord_ be ? GetIterator(_iterable_).
+      1. Repeat,
+        1. Let _next_ be ? IteratorStep(_iteratorRecord_).
+        1. If _next_ is *false*, return _target_.
+        1. Let _nextItem_ be ? IteratorValue(_next_).
+        1. <ins>If Type(_nextItem_) is Tuple, then</ins>
+          1. <ins>Set _nextItem_ to ! ToObject(_nextItem_).</ins>
+        1. <ins>Else</ins> if Type(_nextItem_) is not Object, then
+          1. Let _error_ be ThrowCompletion(a newly created *TypeError* object).
+          1. Return ? IteratorClose(_iteratorRecord_, _error_).
```

Comment:
Looking at the whole spec, `Get()` is used ~140 times on objects. We would need to call it on tuples only ~10 times, so I think it's better to use `GetV` when dealing with "R&T or objects", making it easier to see "this will only be called with objects, thus we can make these assumptions/optimizations" in the majority of cases while reading the spec.

Reply:
What about future edits? It seems like it'd be an easy mistake to make to use Get with an R&T and have it break.

Reply:
It would be as hard/easy as it is to currently accidentally use primitive strings where we expect array-like values.

Reply:
i'm not sure i see that - records are object-like and tuples are array-like, conceptually, and strings being actually arraylike doesn't mean they're conceptually arraylike :-)